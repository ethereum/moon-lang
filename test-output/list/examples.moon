List = zb2rhn9hxhbQp3UYpciYJSJzePFMibeWZaQXJsDV3wGkLmAXX
val = (List "val")
end = (List "end")
a = (val 1 (val 2 (val 3 (val 4 end))))
b = (val 5 (val 6 (val 7 (val 8 end))))
{
  range: (List "range" 0 4)
  sum: (List "sum" a)
  filter: (List "filter" x => (gtn x 2) a)
  concat: (List "concat" a b)
  zipWith: (List "zipWith" (add) a b)
  sum: (List "sum" a)
  mul: (List "mul" a)
  reverse: (List "reverse" a)
  len: (List "len" a)
  match: (List "match" a h => t => h 0)
}
