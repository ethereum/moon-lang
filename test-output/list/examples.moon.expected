{
  range: a => b => (a 0 (a 1 (a 2 (a 3 b))))
  sum: 10
  filter: a => b => (a 3 (a 4 b))
  concat: a => b => 
    (a 1 (a 2 (a 3 (a 4 (a 5 (a 6 (a 7 (a 8 b))))))))
  zipWith: a => b => (a 6 (a 8 (a 10 (a 12 b))))
  mul: 24
  reverse: a => b => (a 4 (a 3 (a 2 (a 1 b))))
  len: 4
  match: 1
}
